### PR TITLE
added APIcalls for manipulation of products

### PR DESCRIPTION
**Pull request by GuptaEesh**
- The code is not refactored.
- It just shows the functionality change

## The comments are not deleted and will be done in next PR ##

### CODE REVIEW

File: src/backend/controllers/CartController.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { Response } from "miragejs";
```
Comment (junaidshaikh-js) [inline src/backend/controllers/CartController.js:1]: What was the need to make changes in the cartController? 

I can see it is just the change in the quotation marks and the same code is added.

File: src/components/composite/Product-cards/Item-Hooks/add-cart.js [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+
+const useAddCartItem = async (product, dispatch, token, setLoader) => {
+    try {
+        setLoader(true)
+        const response = await axios.post(
+            '/api/user/cart',
+            {
+                product,
+            },
+            getConfig(token)
```
Comment (junaidshaikh-js) [inline src/components/composite/Product-cards/Item-Hooks/add-cart.js:28]: Good use of getConfig() function

File: src/components/composite/Product-cards/Item-Hooks/add-wish.js [hunk 3]
```diff
@@ -1,7 +1,29 @@
-const useAddWishItem = (product, dispatchWish) =>
-    dispatchWish({
-        type: 'ADD_TO_WISHLIST',
-        payload: product,
-    })
+// const useAddWishItem = (product, dispatchWish) =>
+import axios from 'axios'
+
+const getConfig = (token) => ({
```
Comment (FarhanMobashir) [inline src/components/composite/Product-cards/Item-Hooks/add-wish.js:4]: You can extract out get config to some helper function and then use it every where Ratner than repeating it

File: src/components/composite/Product-cards/Item-Hooks/add-wish.js [hunk 4]
```diff
@@ -1,7 +1,29 @@
...
+    headers: {
+        authorization: token,
+    },
+})
```
Comment (junaidshaikh-js) [inline src/components/composite/Product-cards/Item-Hooks/add-wish.js:8]: Why not store the function in one file and import it wherever it is needed?

File: src/components/composite/Product-cards/Item-Hooks/remove-cart.js [hunk 5]
```diff
@@ -1,7 +1,32 @@
...
+//     })
+
+// export { useRemoveCartItem }
+import axios from 'axios'
+
+const getConfig = (token) => ({
+    headers: {
+        authorization: token,
+    },
+})
```
Comment (junaidshaikh-js) [inline src/components/composite/Product-cards/Item-Hooks/remove-cart.js:14]: Same as above

**Review by FarhanMobashir — COMMENTED** [review]
Everything is fine and good to go. Just left one suggestion.

**Review by junaidshaikh-js — APPROVED** [review]
Good work. 

I mentioned some points in the comments, other than that everything looks good.

File: src/components/composite/Product-cards/Item-Hooks/add-wish.js (hunk 3 above)
Comment (GuptaEesh) [reply src/components/composite/Product-cards/Item-Hooks/add-wish.js:4]: Yup will do that in the next pr. Thanks for the suggestion

File: src/backend/controllers/CartController.js (hunk 1 above)
Comment (GuptaEesh) [reply src/backend/controllers/CartController.js:1]: I have made one more change if you see closely in update cart handler, I have replaced +=1 with = action.payload according to my UI.

File: src/components/composite/Product-cards/Item-Hooks/add-wish.js (hunk 4 above)
Comment (GuptaEesh) [reply src/components/composite/Product-cards/Item-Hooks/add-wish.js:8]: Yup will do that right now I raised PR just for the backend functionality, now I will do abstraction.

**Comment by GuptaEesh** [comment]
Got 2 reviews. As already told that I have not abstracted code in this pr and refactoring is left and all the comments were around that I am merging the PR.

**Pull request merged**